Assets/BossRoom/Scripts/Server/Game/AIState/AttackAIState.cs

```diff
@@ -122,18 +134,27 @@ private ServerCharacter ChooseFoe()
             return closestFoe;
         }
 
-        private ActionDescription GetCurrentAttackInfo()
+        /// <summary>
+        /// Randomly picks a usable attack. If no actions are usable right now, returns ActionType.None.
+        /// </summary>
+        /// <returns>Action to attack with, or ActionType.None</returns>
+        private ActionType ChooseAttack()
         {
-            m_CurAttackAction = m_AttackActions[Random.Range(0, m_AttackActions.Count)];
+            m_TempListOfAttackActions.Clear();
+            foreach (var actionType in m_AttackActions)
+            {
+                if (m_ActionPlayer.IsReuseTimeElapsed(actionType))
+                {
+                    m_TempListOfAttackActions.Add(actionType);
```

Comment:
To reduce the data complexity of this class, we can remove m_TempListOfAttackActions. We could generate a random index from 0 to (m_AttackActions.Length - 1) and iterate through `m_AttackActions` , checking for `IsReuseTimeElapsed(..)` and return that Action if true, correct? If we have a random start index, we can just iterate through `m_AttackActions.Length` times and wrap around if necessary. If no Actions were found during that iteration, we would return `ActionType.None`.

Reply:
ok